OpenRA.Mods.Common/Pathfinder/PathFinderUnitPathCacheDecorator.cs

```diff
@@ -30,37 +30,47 @@ public PathFinderUnitPathCacheDecorator(IPathFinder pathFinder, ICacheStorage<Li
 			this.cacheStorage = cacheStorage;
 		}
 
-		public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignoreActor)
+		public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignoreActor, BlockedByActor check)
 		{
 			using (new PerfSample("Pathfinder"))
 			{
 				var key = "FindUnitPath" + self.ActorID + source.X + source.Y + target.X + target.Y;
-				var cachedPath = cacheStorage.Retrieve(key);
 
-				if (cachedPath != null)
-					return cachedPath;
+				// Only cache path when transient actors are ignored, otherwise there is no guarantee that the path
```

Comment:
Do we have any idea about how useful this cache actually is? Ping @teinarss.

* A lot has changed since this was added.  If it doesn't do much in practice anymore then we should remove it completely to help simplify the pathfinder mess.
* If we have lots of repeated queries in the same tick then we should cache everything, but expire the non-`BlockedByActor.None` paths at the end of the tick.

Reply:
Im guessing it still is valid for preventing spam click => lag spike.